src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -34,223 +158,176 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
+            [](PTP_CALLBACK_INSTANCE /*callbackInstance*/, PVOID context, PTP_WAIT /*wait*/, TP_WAIT_RESULT /*waitResult*/) {
+                ConptyConnection* const pInstance = reinterpret_cast<ConptyConnection*>(context);
+                pInstance->_ClientTerminated();
+            },
+            this,
+            nullptr));
+
+        SetThreadpoolWait(_clientExitWait.get(), _piClient.hProcess, nullptr);
 
-        throw hresult_not_implemented();
+        _connected = true;
     }
-
-    void ConptyConnection::Resize(uint32_t rows, uint32_t columns)
+    catch (...)
     {
-        rows;
-        columns;
+        const auto hr = wil::ResultFromCaughtException();
+        // TODO GH#2563 - signal a transition into failed state here!
+        (void)hr;
 
-        throw hresult_not_implemented();
+        _disconnectHandlers();
     }
 
-    void ConptyConnection::Close()
+    void ConptyConnection::_ClientTerminated() noexcept
     {
-        // TODO:
-        //      terminate the output thread
-        //      Close our handles
-        //      Close the Pseudoconsole
-        //      terminate our processes
-        throw hresult_not_implemented();
+        if (_closing.load())
+        {
+            // This is okay, break out to kill the thread
+            return;
+        }
+
+        // TODO GH#2563 - get the exit code from the process and see whether it was a failing one.
+        _disconnectHandlers();
     }
 
-    // Function Description:
-    // - Sample function which combines the creation of some basic anonymous pipes
-    //      and passes them to CreatePseudoConsole.
-    // Arguments:
-    // - size: The size of the conpty to create, in characters.
-    // - phInput: Receives the handle to the newly-created anonymous pipe for writing input to the conpty.
-    // - phOutput: Receives the handle to the newly-created anonymous pipe for reading the output of the conpty.
-    // - phPty: Receives a token value to identify this conpty
-    HRESULT _CreatePseudoConsoleAndHandles(COORD size,
-                                           _In_ DWORD dwFlags,
-                                           _Out_ HANDLE* phInput,
-                                           _Out_ HANDLE* phOutput,
-                                           _Out_ HPCON* phPC)
+    void ConptyConnection::WriteInput(hstring const& data)
     {
-        if (phPC == NULL || phInput == NULL || phOutput == NULL)
+        if (!_connected || _closing.load())
         {
-            return E_INVALIDARG;
+            return;
         }
 
-        HANDLE outPipeOurSide;
-        HANDLE inPipeOurSide;
-        HANDLE outPipePseudoConsoleSide;
-        HANDLE inPipePseudoConsoleSide;
+        // convert from UTF-16LE to UTF-8 as ConPty expects UTF-8
+        std::string str = winrt::to_string(data);
+        bool fSuccess = !!WriteFile(_inPipe.get(), str.c_str(), (DWORD)str.length(), nullptr, nullptr);
+        fSuccess;
```

Comment:
Uh. This is `UNREFERENCED_PARAMETER`, just naked. Can we log or report this failure?